From c0084fd6fcf1ceb395d6d915735ae8e4e9a609b8 Mon Sep 17 00:00:00 2001
From: "CyberNeko.AI" <cyberneko.ai@aosc.io>
Date: Sun, 29 Mar 2026 14:14:59 +0800
Subject: [PATCH 2/2] AOSCOS: add autounlock support for kwallet and
 gnome-keyring

Authored by: Lain Yang <fsf@live.com>, originally introduced in
abbs tree as a diff patch.

Forked lightdm to AOSC-Tracking and applied this patch to it.

Signed-off-by: CyberNeko-AI <cyberneko.ai@aosc.io>
---
 data/pam/lightdm | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/data/pam/lightdm b/data/pam/lightdm
index fed8a9b4..4b1f9938 100644
--- a/data/pam/lightdm
+++ b/data/pam/lightdm
@@ -9,6 +9,9 @@ auth      required pam_env.so
 # Use /etc/passwd and /etc/shadow for passwords
 auth      required pam_unix.so
 
+-auth     optional pam_gnome_keyring.so
+-auth     optional pam_kwallet5.so
+
 # Check account is active, change password if required
 account   required pam_unix.so
 
@@ -18,3 +21,6 @@ password  required pam_unix.so
 # Setup session
 session   required pam_unix.so
 session   optional pam_systemd.so
+
+-session  optional pam_gnome_keyring.so auto_start
+-session  optional pam_kwallet5.so
-- 
2.53.0.windows.2

